system/scripts/zone/core/calc_combat.cs

```diff
@@ -192,6 +218,19 @@ public float SCR_CalculateDamage(ICombatEntity attacker, ICombatEntity target, S
 		if (skillHitResult.Result == HitResultType.Crit)
 			damage *= 1.5f;
 
+		// Added effects
+		// Check for Restrain's stun chance, this only activates on normal attacks,
+		// which are skills with ID < 1001
+		if ((int)skill.Id <= 1000 && attacker.TryGetBuff(BuffId.Restrain_Buff, out var restrainBuff))
```

Comment:
For a check like this I would personally rather see a method/property on `Skill`, like `IsNormalAttack`, so we can easily change it down the line if necessary. Because checking a range could break the moment they put some kind of common skill nonsense into it that does damage but isn't considered a normal attack, and then we'd have to change this check everywhere it's done. Additionally it's easier to read and doesn't require a comment.

Reply:
I added one.